File: python/factorial_test_ML.py
import unittest
from parameterized import parameterized
from factorial import factorial  # Import the factorial function

class TestFactorialML(unittest.TestCase):

    def test_factorial_zero(self):
        self.assertEqual(factorial(0), 1)

    @parameterized.expand([(-1,), (-1000,)])
    def test_factorial_negative(self, x):
        self.assertEqual(factorial(x), 0)

    @parameterized.expand([
        (1, 1),
        (2, 2),
        (3, 6)
    ])
    def test_factorial_positive(self, x, result):
        self.assertEqual(factorial(x), result)

if __name__ == '__main__':
    unittest.main()

File: java/FactorialTestML.java
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.*;
import org.junit.jupiter.params.*;
import org.junit.jupiter.params.provider.*;

public class FactorialTestML {

  @Test
  public void testFactorialZero() {
    assertEquals(1L, Factorial.factorial(0));
  }
 
  @ParameterizedTest
  @ValueSource(ints = { -1, -1000 })
  public void testFactorialNegative(int x) {
    assertEquals(0L, Factorial.factorial(x));
  }

  @ParameterizedTest
  @CsvSource({ "1,1", "2,2", "3,6" })
  public void testFactorialPositive(int x, long result) {
    assertEquals(result, Factorial.factorial(x));
  }
}
